docs/proposals/1374-multi-cluster-inference/README.md

```diff
@@ -0,0 +1,401 @@
+# Multi-Cluster Inference Pooling
+
+Author(s): @danehans, @bexxmodd, @robscott
+
+## Proposal Status
+
+ ***Draft***
+
+## Summary
+
+An Inference Gateway (IG) provides efficient routing to LLM workloads in Kubernetes by sending requests to an Endpoint Picker (EPP) associated with
+an [InferencePool](https://gateway-api-inference-extension.sigs.k8s.io/api-types/inferencepool/) and routing the request to a backend model server
+based on the EPP-provided endpoint. Although other multi-cluster inference approaches may exist, this proposal extends the current model to support
+multi-cluster routing so capacity in one cluster can serve traffic originating in another cluster or outside the cluster.
+
+### Why Multi-Cluster?
+
+GPU capacity is scarce and fragmented. Many users operate multiple clusters across regions and providers. A single cluster rarely satisfies peak or
+sustained demand, so a prescribed approach is required to share GPU capacity across clusters by:
+
+- Exporting an InferencePool from a source (“exporting”) cluster.
+- Importing the exported InferencePool into one or more destination (“importing”) clusters with enough detail for IGs to route requests to the associated
+  remote model server Pods.
+
+### Goals
+
+- Enable IGs to route to a group of common model server Pods, e.g. InferencePools, that exist in different clusters.
+- Align the UX with familiar [Multi-Cluster Services (MCS)](https://multicluster.sigs.k8s.io/concepts/multicluster-services-api/) concepts (export/import).
+- Keep the API simple and implementation-agnostic.
+
+### Non-Goals
+
+- Managing DNS or automatic naming.
+- Over-specifying implementation details to satisfy a single approach to multi-cluster inference.
+
+## Design Proposal
+
+The Multi-Cluster Inference Pooling (MCIP) model will largely follow the Multi-Cluster Services (MCS) model, with a few key differences:
+
+- DNS and ClusterIP resolution will be omitted, e.g. ClusterSetIP.
+- A separate export resource will be avoided, e.g. ServiceExport, by inlining the concept within InferencePool.
+
+An InferencePoolImport resource is introduced that is meant to be fully managed by a controller. This resource provides the information
+required for IGs to route LLM requests to model server endpoints of an InferencePool in remote clusters. How the IG routes the request to the remote
+cluster is implementation-specific.
+
+### Routing Modes
+
+The proposal supports the following routing modes:
+
+- Endpoint Mode: An IG of an importing cluster routes to endpoints selected by the EPP of the exported InferencePool. Pod and Service network connectivity
+  MUST exist between cluster members.
+- Parent Mode: An IG of an importing cluster routes to parents, e.g. Gateways, of the exported InferencePool. Parent connectivity MUST exist between cluster
```

Comment:
This is probably too in the weeds for the proposal, but one of the considerations that I think will be relevant for parent mode is the potential for double L7 processing (e.g. auth policy, routing loops, etc.). The local cluster (hosting the first-hop gateway) has to do L7 routing (including potentially BBR) in order to know which InferencePool to send to (after e.g. applying an HTTPRoute). Then, once traffic goes to the remote gateway (i.e. the remote InferencePool parent), it also has to do L7 processing (at least Envoy implementations do) in order to call out to the EPP to figure out which InferencePool to route to.

Reply:
@keithmattix ^ is something that I've been thinking about as we have iterated on this initial proposal. If implementations start to rally around a certain parent routing pattern, we can definitely update the proposal to define these details.